docs/proposals/11-scheduling.md

```diff
@@ -0,0 +1,162 @@
+
+```protobuf
+rpc ListReservations(ListReservationsRequest) returns (ListReservationsResponse) {};
+rpc CreateReservation(CreateReservationRequest) returns (CreateReservationResponse) {};
+rpc DeleteReservation(DeleteReservationRequest) returns (DeleteReservationResponse) {};
+```
+
+
+### Detailed flow
+
+The flow to assign a `Machine` to a `Pool` consists of 3 Phases:  
+
+#### 1. Reservation Flow
+1. If `.spec.machinePoolRef` is not set, the scheduler creates a `Reservation` that includes the required resources.
+2. `poollet`s which match `.spec.pools` of `Reservation` pick it up and broker it one layer down
+3. `provider` evaluates the `Reservation` and sets its state to `Accepted` or `Rejected`, which is then 
+   propagated up the hierarchy.
+
+#### 2. Scheduling Flow
+1. On every layer, the scheduler uses a `Reservation` corresponding to a `Machine` to select a `MachinePool`. It 
+   will pick one of the `Accepted` pools of the `Reservation.status.pools` and updates the `Machine.spec.
+   machinePoolRef`.
+2. `poollet` picks up the `Machine` since `.spec.machinePoolRef` is set
+3. Once a `Machine` reaches a `provider` with a relating `Reservation`, the `Reservation` will be replaced through
+   the `Machine`. `Reservation` status will be updated to `Bound` and `poollet`s pulls the status up until to the 
+   root `Reservation`.
+
+#### 3.  Cleanup Flow
+1. The scheduler deletes root `Reservation` if `Machine` has `.spec.machinePoolRef` and if `Reservation` has a 
+   `Bound` state.
+
+### Advantages
+- In case of a partial outage/network issues `Machine`s can be placed on other `Pool`s
+- `Pool`s do not leak resource information, owner of resources decides if `Reservation` can be fulfilled (less complexity for over provisioning)
+- `Reservation`s can be used to block resources without creating a `Machine`
+
+### Disadvantages
+- Increases complexity in the scheduling by introducing a new resource (Reservation) and the scheduler flow has to be extended.
+
+## Alternatives
+
+Another solution: The providers (in the lowest layer) announce there resources to a central cluster. In these clusters a `shallow` pool represent the actual compute pool. In fact the problem of scheduling across multi-levels is transformed in a one-layer scheduling decision.  Pools between the root cluster and the leaf clusters are announced but only to represent the hierarchy and not for the actual scheduling.
+
+If a `Machine` is created, a controller creates a related `Machine` in the central cluster where the `Pool` is being picked. The `Pool` is synced back and the`.spec.machinePoolRef` being set. The `poollet` syncs the `Machine` one layer down. In every subsequent layer the path is being looked up in the central cluster until the provider is hit.
+
+### Advantages
+- Scheduling in central place is simpler
+- Availability/failure zones can be modeled in central cluster
+- (Scheduling decision takes potentially less time?)
+
+### Disadvantages
```

Comment:
another point here is that to model this correctly a poollet has to be aware of if its a leave or a branch poollet. This is required here to correctly model the topology as a branch poollet needs to announce the grouped machinepools under its own aggregated pool.

Reply:
I think it's covered with this point, right? `- Layered structure of hierarchy needs to be duplicated at central place`

Reply:
the importance is that it requires a change in the poollet / broker design that could potentially break the current approach for those components.